### PR TITLE
[bug] safetykernel: fail closed on unreadable result pointer in gRPC CheckOutput (CRD-3, task-74a83bec)

### PR DESCRIPTION
**Pull request by yaront1111**
﻿## Summary

Fixes CRD-3 / Moe task `task-74a83bec`: the gRPC `CheckOutput` path now fail-closes when Redis result-pointer dereference fails with a non-`redis.Nil` error.

## Root cause

`EvaluateOutput` already treated unreadable result pointers as a definitive quarantine decision, but `CheckOutput` called `contentForScan`, logged the Redis read error, then fell through to empty/error-message content. That preserved the default ALLOW response in the production gRPC path.

## Before/after divergence table

| Branch | Before | After |
| --- | --- | --- |
| Non-`redis.Nil` result-pointer read error | `EvaluateOutput`: QUARANTINE + critical `pointer_unreadable`; `CheckOutput`: logs then may return ALLOW | Both paths return QUARANTINE + critical `pointer_unreadable` |
| `redis.Nil` missing key | Falls through to no-content/error-message fallback | Unchanged: falls through; allow-by-default unless another rule matches |
| Nil result client | Warn/fallback behavior | Unchanged |
| Invalid pointer | Warn/fallback behavior | Unchanged |
| Inline output content | Truncate + scan | Unchanged |
| Policy disabled | ALLOW before acquisition | Unchanged |

## Reachability honesty note

The scheduler path has a narrow window because `OutputSafetyClient.EvaluateOutput` may pre-load result content before calling `CheckOutput`, but any direct gRPC caller (and scheduler paths where content cannot be preloaded into the request) hit this fail-open divergence cleanly. Returning a nil gRPC error is deliberate so callers persist the result as a quarantine decision instead of treating it as a transport failure/fail-open candidate.

## Tests / verification

- RED verified pre-fix: `TestCheckOutput_FailClosedOnResultGetError` failed with `expected result pointer read error to fail closed, got OUTPUT_DECISION_ALLOW`.
- `go test -count=3 ./core/controlplane/safetykernel/` → `ok ... 15.743s`
- `go vet ./core/controlplane/safetykernel/...` → EXIT=0
- `go build ./...` → EXIT=0

## Merge order

**MERGE-ORDER:** task-5f641afe / CRD-4 also touches `contentForScan`/output safety behavior and must rebase on this PR after it lands.

## Summary by CodeRabbit

* **Bug Fixes**
  * Fail-closed behavior for truncated outputs when sensitive scanners are present; REDACT on truncated content now escalates to quarantine.
  * Fail-closed quarantine when output pointer/read errors occur instead of allowing the response.

* **Documentation**
  * Added scan-size limit details (2 MiB) and guidance on fail-closed behavior for truncated sensitive content.

* **Tests**
  * Added coverage for truncation edge cases and pointer/read failure scenarios.

### CODE REVIEW

**Comment by yaront1111** [comment]
@coderabbitai review

File: core/controlplane/safetykernel/output_policy.go [hunk 1]
```diff
@@ -187,25 +187,37 @@ func (s *server) EvaluateOutput(ctx context.Context, req *OutputEvaluateRequest)
...
+		// Truncated content with sensitive scanners fails closed: a REDACT
+		// decision can only cover the scanned head, not the unscanned tail.
+		if contentTruncated && hasSensitiveScanners && outputDecisionWeakerThanQuarantine(rule.decision) {
+			findings = append(findings, outputTruncatedFailClosedFinding())
+			resp.Decision = "quarantine"
+			resp.Reason = outputTruncatedFailClosedReason
+			resp.RuleID = rule.id
+			resp.Findings = findings
+			return resp, nil
+		}
```
[the remark this answers is not in the document]
Comment (yaront1111) [reply core/controlplane/safetykernel/output_policy.go:205]: Verified against the contract and pushing back on the suggested diff: scoping the in-loop escalation to `outputRuleUsesSensitiveScanner(rule)` reintroduces the leak this PR exists to close. Counter-example: policy = [keyword rule (redact), secret rule (quarantine)], secret past the scan cap, keyword in the head. With a full scan the secret rule can quarantine; with per-rule scoping the keyword rule matches first and releases REDACT while the tail was never scanned by the secret scanner — truncation producing a *weaker* outcome than full scanning, which is exactly what the fail-closed mode prevents. The docs line about non-sensitive rules keeping partial-scan behavior is snapshot-scoped ("outputs governed only by non-sensitive rules"), matching `TestCheckOutput_TruncatedNonSensitiveRules_Unchanged`.

The missing-regression-test point was valid: added `TestCheckOutput_TruncatedMixedPolicy_NonSensitiveMatchStillFailsClosed` in fbd78379 pinning quarantine for the mixed case.

A genuine over-blocking dimension does exist cross-topic (a sensitive rule scoped to an unrelated topic still forces escalation here); the precise fix is applicability-scoped (topic/tenant-matched sensitive rules), not matched-rule-scoped — noted as a follow-up.

**Review by bigboycoder1 — APPROVED** [review]
